### PR TITLE
feat: implement fieldttl for the hash family

### PR DESCRIPTION
**Pull request by romange**
Fix some corner case bug with set family as well.

### CODE REVIEW

File: src/server/hset_family.cc [hunk 1]
```diff
@@ -1201,4 +1204,25 @@ StringMap* HSetFamily::ConvertToStrMap(uint8_t* lp) {
 
   return sm;
 }
+
+// returns -1 if no expiry is associated with the field, -3 if no field is found.
```
Comment (dranikpg) [inline src/server/hset_family.cc:1208]: very small nit: better we returned an optional<uint32_t> here (field not found), and did the casting to -3 in the command handler itself

**Review by dranikpg — DISMISSED** [review]

File: src/server/hset_family.cc [hunk 2]
```diff
@@ -518,12 +521,12 @@ OpResult<string> OpGet(const OpArgs& op_args, string_view key, string_view field
 
   DCHECK_EQ(pv.Encoding(), kEncodingStrMap2);
   StringMap* sm = GetStringMap(pv, op_args.db_cntx);
-  sds val = sm->Find(field);
+  auto it = sm->Find(field);
 
-  if (!val)
+  if (it != sm->end())
     return OpStatus::KEY_NOTFOUND;
```
Comment (dranikpg) [inline src/server/hset_family.cc:527]: Here's the bug that causes hset test failures : it should be `it == sm->end()`

File: src/server/hset_family.cc (hunk 1 above)
Comment (romange) [reply src/server/hset_family.cc:1208]: why it is better on your opinion? we need to return multiple statuses: -3, -1.

File: src/server/hset_family.cc (hunk 2 above)
Comment (romange) [reply src/server/hset_family.cc:527]: thanks

**Review by dranikpg — APPROVED** [review]

**Pull request merged**